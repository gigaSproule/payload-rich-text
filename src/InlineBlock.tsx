import { CSSProperties } from "react";
import type { Options } from "./types";

export type InlineBlockData = {
  type: "inlineBlock";
  fields: object;
  version: number;
};

export type Props = {
  data: InlineBlockData;
  options?: Options;
  style?: CSSProperties;
};

export const InlineBlock = ({ data, options, style }: Props) => {
  return options?.inlineBlock ? options.inlineBlock(data, style) : undefined;
};
