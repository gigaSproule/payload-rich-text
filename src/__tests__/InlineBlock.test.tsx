import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../InlineBlock";
import { InlineBlock } from "../InlineBlock";
import { createInlineBlockData } from "./helpers/createInlineBlockData";

describe("InlineBlock", () => {
  it("should not render anything if no options provided", () => {
    const inlineBlockData: Props["data"] = createInlineBlockData();
    const { container } = render(<InlineBlock data={inlineBlockData} />);
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with horizontalRule", () => {
    const inlineBlockData: Props["data"] = createInlineBlockData();
    const { container } = render(
      <InlineBlock
        data={inlineBlockData}
        options={{
          inlineBlock: (inlineBlock, style) => {
            return (
              <div>
                <span>Custom InlineBlock - {JSON.stringify(inlineBlock)}</span>
                {JSON.stringify(style)}
              </div>
            );
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
