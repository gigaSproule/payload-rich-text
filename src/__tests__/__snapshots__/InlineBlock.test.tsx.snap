// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineBlock should not render anything if no options provided 1`] = `<div />`;

exports[`InlineBlock should render custom render if options provided with horizontalRule 1`] = `
<div>
  <div>
    <span>
      Custom InlineBlock - 
      {"type":"inlineBlock","version":1,"fields":{"some":"custom","block":"fields"}}
    </span>
  </div>
</div>
`;
